### PR TITLE
Change inproc transport to set trailing_metadata_available.

### PR DESCRIPTION
**Pull request by markdroth**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                         FILE SIZE
 ++++++++++++++ GROWING                                           ++++++++++++++
  +0.8%     +96 src/core/ext/transport/inproc/inproc_transport.cc     +96  +0.8%
      +1.6%     +47 op_state_machine                                      +47  +1.6%
      +2.3%     +32 fail_helper_locked                                    +32  +2.3%
      +0.7%     +16 perform_stream_op                                     +16  +0.7%
      +0.5%      +1 [Unmapped]                                             +1  +0.5%

 -------------- SHRINKING                                         --------------
  [ = ]       0 [None]                                               -112  -0.0%

  +0.0%     +96 TOTAL                                                 -16  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: src/core/ext/transport/inproc/inproc_transport.cc [hunk 1]
```diff
@@ -670,6 +678,12 @@ static void op_state_machine(grpc_exec_ctx* exec_ctx, void* arg,
           nullptr);
       s->recv_initial_md_op->payload->recv_initial_metadata
           .recv_initial_metadata->deadline = s->deadline;
+      if (s->recv_initial_md_op->payload->recv_initial_metadata
+              .trailing_metadata_available != nullptr) {
+        *s->recv_initial_md_op->payload->recv_initial_metadata
+            .trailing_metadata_available =
+                other != nullptr && other->send_trailing_md_op != nullptr;
```
Comment (vjpai) [inline src/core/ext/transport/inproc/inproc_transport.cc]: Can I request a set of parens around the contents of line 685?  Line 683-684 is so long that it is easy to misparse this.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/ext/transport/inproc/inproc_transport.cc (hunk 1 above)
Comment (markdroth) [reply src/core/ext/transport/inproc/inproc_transport.cc]: Done.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                         FILE SIZE
 ++++++++++++++ GROWING                                           ++++++++++++++
  +0.8%     +96 src/core/ext/transport/inproc/inproc_transport.cc     +96  +0.8%
      +1.6%     +47 op_state_machine                                      +47  +1.6%
      +2.3%     +32 fail_helper_locked                                    +32  +2.3%
      +0.7%     +16 perform_stream_op                                     +16  +0.7%
      +0.5%      +1 [Unmapped]                                             +1  +0.5%

 -------------- SHRINKING                                         --------------
  [ = ]       0 [None]                                               -112  -0.0%

  +0.0%     +96 TOTAL                                                 -16  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by markdroth** [comment]
Known failures: #12917 #13643

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                         FILE SIZE
 ++++++++++++++ GROWING                                           ++++++++++++++
  +0.8%     +96 src/core/ext/transport/inproc/inproc_transport.cc     +96  +0.8%
      +1.6%     +47 op_state_machine                                      +47  +1.6%
      +2.3%     +32 fail_helper_locked                                    +32  +2.3%
      +0.7%     +16 perform_stream_op                                     +16  +0.7%
      +0.5%      +1 [Unmapped]                                             +1  +0.5%

 -------------- SHRINKING                                         --------------
  [ = ]       0 [None]                                               -112  -0.0%

  +0.0%     +96 TOTAL                                                 -16  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**